### ISSUE TITLE
Universalize the ConditionalFields function.

### ISSUE THREAD

**Issue by Nicholas-Cardot** [warfare-plugins/social-warfare#463]
So we want to be able to use the conditonalFields function on the admin options page, the widget options page, and eventually on the post editor page. As such, we want to migrate it into the admin.js file so that it is loaded everywhere across the admin side of WordPress.

Currently, the "name" attribute on the widgets page is dynamically generated and is different each time depending on the number of widgets that are active. Since the function targets that "name" attribute, this is problematic. Instead, we should switch the plugin to use a class based system or instead, create a custom attribute of some sort. We can then add this class or attribute to each field in our forms and target that instead as it will be static.

We will need to add this attribute to the fields to be output on the admin option page via the loop in admin-options-page.php. We will need to add that attribute to the fields in widgets.php. And eventually, when we go to apply this to the post editor options, we will add the attribute to those fields as well.

- [x] Make the function look for the "data-swp-name" attribute instead of the "name" attribute.
- [x] Add the "data-swp-name attribute" to all of the field generators in admin-options-page.php
- [x] Add the "data-swp-name attribute" to all of the fields in widgets.php. 
- [x] Convert existing custom attributes to `data-` versions of the same thing
- [x] Update each instance targeting the custom attributes to grab the dataset instead
- [x] Switch all other fields in widgets.php to use the new JS dependancy function, not a PHP conditional.
- [x] Make sure the function runs on Dom loaded and whenever a field with a "data-dep" attribute is changed.
- [x] Add a new docblock to the top of the admin.js file to begin documenting the file and it's history.
- [x] Add a new docblock immediately above this function to document this function and it's history.

Of course, those docblocks will be removed from admin.min.js upon the minification process, but it's important that we have them in there for internal record keeping/historical purposes.

*Important*: Make sure that the `data-dep` and `data-dep-value` attributes go on the element that you want to have shown/hidden, which in every single case will be a parent/wrapper class that encapsulates the entire options (e.g. the label, the input, etc.) so that the whole thing hides.

**Comment by ckmahoney** [comment]
Is there a motivation of using `<p dep="x" dep-val="y"/>` rather than `<p data-dep="x" data-dep-val="y"/>`? The plugin works with the existing custom attributes "dep" and "dep-val". We could have access to Javascript features by working with the HTML5 standard data attribute though. 
```
<script>
val = document.getElementById("target").dataset("dep-val");
$(parentEl).addClass("visible");
</script>
```

jQuery makes it easy particularly easy to fetch data: `$(el).data("dep-val")`. 

But, I suppose this is all about the same as a call to `$el).attr("dep-val")`. The key difference being how the DOM interacts with it. Custom attributes are not valid HTML, so the DOM just ignores them (and we take advantage of that with Javascript). Data attributes are valid, so the DOM parses them but does not do anything with them aside from letting JS and CSS know they exist.

**Comment by Nicholas-Cardot** [comment]
Ah, good point. I just abritrarily made those attributes up. Totally forgot about the whole data- thing. So yeah, let's definitely add those. Just make sure to also add them to the loop on options-page.php.

**Comment by ckmahoney** [comment]
Okay. I have updated the checklist above to reflect that.

**Comment by ckmahoney** [comment]
The widget seems to save the data with calls do admin-ajax. Right now the hiding/showing is not showing up correctly. My eyes are not seeing what is wrong right now, so I am stepping away from this task. 

**Todo**

- [ ] Correct dependencies are shown (or hidden) on initial page load
- [ ] Correct dependencies displayed on widget save

**Comment by Nicholas-Cardot** [comment]
Since this function is now in the global scope, we should probably add our vendor prefix to it and to all calls to it.
